### PR TITLE
Fix `NamespaceCache` usage in `System.Private.Xml.Linq`

### PR DESCRIPTION
**Pull request by hamarb123**
See https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XLinq.cs#L181-L193 - it's intended to be mutated on `Get` so that it can cache the `XNamespace` object - the fields being marked `readonly` completely block all caching.

### CODE REVIEW

File: src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XContainer.cs [hunk 1]
```diff
@@ -880,8 +880,8 @@ internal async Task ReadContentFromAsync(XmlReader r, LoadOptions o, Cancellatio
 
         private sealed class ContentReader
         {
-            private readonly NamespaceCache _eCache;
-            private readonly NamespaceCache _aCache;
+            private NamespaceCache _eCache;
+            private NamespaceCache _aCache;
```
Comment (stephentoub) [inline src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XContainer.cs:884]: It looks like this occurred as part of moving these from being locals to being fields as part of https://github.com/dotnet/runtime/commit/d88ba5b82dd0d1164869fe6d5fbcd41c7a62f26c#diff-b2b608017bdbacf5e4484368d6e1fcb03a9ea3b68f4aece41da253f72826acf9R908. Would they be better again as locals? I'm not sure why they were moved to be fields, but they're increasing the size of the ContentReader object by being here.
cc: @krwq

**Comment by krwq** [comment]
@hamarb123 please see @stephentoub's comments

**Comment by hamarb123** [comment]
> @hamarb123 please see @stephentoub's comments

So, we want to move them to locals and pass them by reference to the methods?
Otherwise, they're just going to be called once making the caching pointless (like it is currently due to `readonly`) as all the current uses I looked at only call it once per method.

**Comment by hamarb123** [comment]
So, it seems like previously all the stuff was done in `ReadContentFrom` - but now it's done in multiple sub-funtions, seemingly to support async. Previously the cache was declared in `ReadContentFrom` and used throughout the whole invocation. It seems like the same cache is still supposed to be used throughout the whole iteration, except that it was accidentally marked `readonly` (which accidentally broke the caching) - but it was moved to be a field, not a local, due to being easier to that being easier to work with `async` presumably. It looks like we could make it a local - for the sync version that would save a few bytes of allocation, and for the async version it would move the allocation - and just pass it by byref to the sync method that uses it (I assume we can do this in async?).

Please confirm the preferred approach @krwq @stephentoub

**Comment by stephentoub** [comment]
Ok, let's keep them as fields for now but revisit subsequently.

**Review by stephentoub — APPROVED** [review]

**Pull request merged**